File: python_src/picard_iteration.py
import numpy as np

def picard_iteration(iterative, x_0, epsilon):
    k = 1
    x_cur = x_0
    
    while True:
        x_next = iterative(x_cur)
        
        if abs(x_next - x_cur) < epsilon:
            print(f"{k} iteration: {x_next}")
            break
        
        x_cur = x_next
        print(f"{k} iteration: {x_cur}")
        k += 1
        
def iterative(x):
    return np.exp(-x)

picard_iteration(iterative, 0.5, 1e-5)

File: matlab_src/picard_iteration.m
function result = picard_iteration(g, x_0, tolerance)
    k = 1;
    x_current = x_0;
    while true
        x_next = g(x_current);

        if abs(x_next - x_current) < tolerance
            result = x_next;
            fprintf('%d result: %.14f\n', k, result);
            return;
        end

        x_current = x_next;
        fprintf('%d result: %.14f\n', k, x_current);
        k = k + 1;
    end
end
